File: springboot-jwt/src/main/java/com/grepp/spring/app/model/auth/entity/RefreshToken.java
package com.grepp.spring.app.model.auth.entity;

import java.util.UUID;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;

@Getter @Setter
@RedisHash(value = "refreshToken", timeToLive = 3600 * 24 * 7)
public class RefreshToken {

    @Id
    private String email;
    private String token= UUID.randomUUID().toString();

    public RefreshToken(String email) {
        this.email = email;
    }
}

File: grepp-gateway/src/main/kotlin/com/grepp/spring/token/entity/RefreshToken.kt
package com.grepp.spring.token.entity

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.index.Indexed
import java.util.*

@RedisHash(value = "refreshToken", timeToLive = 3600 * 24 * 7)
class RefreshToken(
    @Id
    val id: String = UUID.randomUUID().toString(),
    var email: String,
    @Indexed
    var accessTokenId: String,
    var token: String = UUID.randomUUID().toString()
){

}
